Improve pruning documentation (#3444)
NuGet/docs.microsoft.com-nuget: docs/consume-packages/Package-References-in-Project-Files.md

## PrunePackageReference

The .NET Runtime is constantly evolving, with performance improvements and new APIs each release.

> There is a lot of functionality that's available within the runtime, but also as packages, such as [System.Text.Json](https://www.nuget.org/packages/System.Text.Json).

Reviewer: very opinionated: It looks like my suggestion is not shorter than what's already written here, but to me "There is a lot of functionality that's available within the runtime" reads funny to me.

I'm thinking something along the lines:

```suggestion
New features added to .NET sometimes is also provided as packages, so that developers using older target frameworks can use the library, such as [System.Text.Json](https://www.nuget.org/packages/System.Text.Json).
```

Author: ```suggestion
New features added to .NET sometimes are also provided as packages, so that developers using older target frameworks can use the library, such as [System.Text.Json](https://www.nuget.org/packages/System.Text.Json).
```

### How PrunePackageReference works

When a package is specified to be pruned during restore, it is removed from the dependency graph.

> When a package is pruned, there is a detailed verbosity message indicating that the package has been removed for the given target framework.

Reviewer: very opinionated: 

```suggestion
When a package is pruned, there is a message, visible at detailed verbosity, indicating that the package has been removed for the given target framework.
```

> For transitive packages, meaning packages that are referenced by other packages or projects, the packages are not downloaded and do not appear in any of the outputs of NuGet.

Reviewer: ```suggestion
For transitive packages, meaning dependencies of other packages or projects, the packages are not downloaded and do not appear in any of the outputs of NuGet.
```
